x-pack/plugins/upgrade_assistant/server/routes/ml_snapshots.ts

```diff
@@ -81,7 +81,7 @@ const verifySnapshotUpgrade = async (
         const regex = /(?<=\[).*?(?=\])/g;
         const matches = snapshotDeprecation.message.match(regex);
 
-        if (matches?.length === 2) {
+        if (matches?.length === 3) {
```

Comment:
I know it is not part of this PR but any chance we could add a comment on top of the `regex` to explain exactly what we are looking for. Then it would be easier to understand why we needed to change this line.